### DISCUSSION TITLE
Preview all images being generated when batch size is > 1

### DISCUSSION THREAD

**Discussion by shripadk** [comfyanonymous/ComfyUI#2505] [Q&A]
As the question suggests currently preview is being shown for only 1 image when batch size is set to "n". Is there a way to show preview being generated for all "n" images? I want this specifically while calling the API. Or will only preview of 1 image be shown?

**Comment by blepping** [comment]
@shripadk you can give https://github.com/blepping/ComfyUI-bleh a try. note: barely tested at this point, seems to work for me. i am not sure if you would need to do something special when using the api, but you can potentially take that code and modify it if you need to

**Reply by shripadk** [reply]
Thank you! I'll give it a try and get back to you in a few hours! Pretty sure I'll be able to use the API without issues.

**Reply by blepping** [reply]
no rush or anything, but did it do what you were looking for? can't promise anything but if not i am open to suggestions for improvements.

**Reply by hansolocambo** [reply]
> no rush or anything [...]

but could you please post workflows, or screenshots of workflows for your "bleh" nodes. I'm sure you have a precise idea of how to use them. But when one can't figure that, out of the blue, and neither Gemini nor ChatGPT can help... it's a bit of a pain on the user side. 
Not being able to preview more than the first image of a batch sucks. I can't seem to find anything around to do that except your nodes... but now I'm stuck because your github is so empty of useful info. 
Please consider updating a bit your page, for dummies. Thanks. 

P.S: I don't really get how you and so many other people posting nodes, can spend a crazy amount of time coding nodes. And so little time explaining with informative screenshots: HOW to use them...

**Reply by blepping** [reply]
@hansolocambo 

> Not being able to preview more than the first image of a batch sucks. I can't seem to find anything around to do that except your nodes... but now I'm stuck because your github is so empty of useful info.

I have to say, I'm kind of confused by your criticism here. There are _pages_ of documentation for ComfyUI-bleh and most of the nodes have several paragraphs of description devoted to explaining how they work. The previewer part is especially well documented and there's information about all the configuration settings related to previewing, links to download preview models, etc. So, uh... I guess I'd have to respond with "What are you talking about?"

As for previews specifically, ComfyUI-bleh defaults to showing previews for multiple batch items (and also video frames, if you're doing video generation). There isn't anything special you need to do to enable that, it's not a node and there's nothing to screenshot.

It's possible you have some other custom node pack interfering with my previewer hooking. For example, VideoHelperSuite with animated previews turned on will install its own previewer and potentially overwrite mine. There is a `BlehEnsurePreviewer` node you can use to force the Bleh previewer to be used. It's also possible that something went wrong and `ComfyUI-bleh` didn't even get loaded. You can determine if that's the case by checking your console startup messages or making sure you have the nodes ComfyUI-bleh provides in your node list.

> I'm sure you have a precise idea of how to use them. But when one can't figure that, out of the blue, and neither Gemini nor ChatGPT can help... it's a bit of a pain on the user side.

You could try creating an issue or discussion in my repo and asking?

> P.S: I don't really get how you and so many other people posting nodes, can spend a crazy amount of time coding nodes. And so little time explaining with informative screenshots: HOW to use them...

I tried explaining with informative _words_, and I actually spent quite a bit of time doing that. You should also bear in mind that open source projects are _mostly_ motivated by people wanting some feature or functionality themselves and then sharing it in case it is useful to other people as well. In other words, I didn't just make something for other people out of pure altruism.

It's an open source project so if you think the documentation could be improved or adding screenshots would be useful then you could also spend some of your own time/energy contributing.
